src/services/blockchain/ethereum/ethereum-blockchain-service.ts

```diff
@@ -41,42 +43,83 @@ export default class EthereumBlockchainService implements BlockchainService {
     }
 
     await this.provider.getNetwork();
-    logger.Imp('Connected to ' + config.blockchain.connectors.ethereum.network + ' blockchain.');
+    logger.Imp("Connected to " + config.blockchain.connectors.ethereum.network + " blockchain.");
   }
 
   /**
    * Sends transaction with root CID as data
    */
   public async sendTransaction(rootCid: CID): Promise<Transaction> {
     const wallet = new ethers.Wallet(config.blockchain.connectors.ethereum.account.privateKey, this.provider);
+    const walletBalance = await this.provider.getBalance(wallet.address);
+    logger.Imp(`Current wallet balance is ` + walletBalance);
 
-    const rootStrHex = rootCid.toString('base16');
-    const hexEncoded = '0x' + (rootStrHex.length % 2 == 0 ? rootStrHex : '0' + rootStrHex);
+    const rootStrHex = rootCid.toString("base16");
+    const hexEncoded = "0x" + (rootStrHex.length % 2 == 0 ? rootStrHex : "0" + rootStrHex);
     logger.Imp(`Hex encoded root CID ${hexEncoded}`);
 
     const { network } = config.blockchain.connectors.ethereum;
     logger.Imp(`Sending transaction to Ethereum ${network} network...`);
 
-    const txData = {
+    const txData: TransactionRequest = {
       to: wallet.address,
-      data: hexEncoded,
+      data: hexEncoded
     };
 
-    if (config.blockchain.connectors.ethereum.overrideGasConfig === true) {
-      Object.assign(txData, {
-        gasLimit: +config.blockchain.connectors.ethereum.gasLimit,
-        gasPrice: BigNumber.from(config.blockchain.connectors.ethereum.gasPrice),
-      });
+    let { overrideGasConfig } = config.blockchain.connectors.ethereum;
+    if (typeof overrideGasConfig === "string") {
+      overrideGasConfig = overrideGasConfig as string === 'true'
     }
+    if (!overrideGasConfig) {
+      txData.gasPrice = await this.provider.getGasPrice();
+      logger.Info('Estimated Gas price: ' + txData.gasPrice.toString());
```

Comment:
This needs to happen inside of the retry loop. Otherwise we could run into trouble when resubmitting and the gasPrice has increased.

Reply:
yep, makes sense 👍